Model/ResourceModel/Events/Collection.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Klaviyo\Reclaim\Model\ResourceModel\Events;
+
+use Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection;
+
+class Collection extends AbstractCollection
+{
+    protected $_idFieldName = 'id';
+
+    /**
+     * Define resource model
+     *
+     * @return void
+     */
+    protected function _construct()
+    {
+        $this->_init( 'Klaviyo\Reclaim\Model\Events', 'Klaviyo\Reclaim\Model\ResourceModel\Events' );
+    }
+
+    public function getEventsToUpdate()
+    {
+        $eventsCollection = $this->addFieldToSelect( ['id','event','payload','user_properties'] )
+            ->addFieldToFilter( 'status','New' )
+            ->addOrder( 'created_at', self::SORT_ORDER_ASC )
+            ->setPageSize( 500 );
+
+        return $eventsCollection;
+    }
+
+    public function getIdsToDelete()
+    {
+        $now = new \DateTime('now');
+        $date = $now->sub( new \DateInterval('P2D') )->format('Y-m-d H:i:s');
+
+        $tableName = $this->getMainTable();
+
+        $idsToDelete = $this->getConnection()->fetchAll( "select id from (
+                                                            select id, timestampdiff(day, created_at, \"$date\") as row_age_in_days
+                                                            from $tableName
+                                                            where status = '".self::MOVED."'
+                                                            having row_age_in_days > 2
+                                                          ) as age_of_rows;"
+        );
+
+        return $idsToDelete;
+    }
```

Comment:
Knowing that we've got the cron logic coming in another PR, wondering if this shouldn't live there instead. I'm looking at some other cron cleanup logic e.g.
https://github.com/magento/magento2/blob/86dc050f2669617d1b34bc89a4a14522f008bf76/app/code/Magento/Sales/Model/CronJob/CleanExpiredOrders.php#L56

Do we want to be consistent and follow the pattern Magento uses and build that query in the cron class?

I could be way off base here in which case just say so.

Reply:
It is cleaner to have this live in a AbstractClass(KlaviyoCollection) which extends the Magento Abstract collection and all Klaviyo collections inherit methods from this AbstractClass, see feedback commit